### PR TITLE
Feature/vdx 316  deferred credential support

### PR DESCRIPTION
**Pull request by nklomp**
Add deferred credential support

### CODE REVIEW

File: packages/client/lib/CredentialRequestClient.ts [hunk 1]
```diff
@@ -19,25 +21,57 @@ import { isValidURL, post } from './functions';
...
+  proofInput: ProofOfPossessionBuilder<DIDDoc> | ProofOfPossession,
+  opts: {
+    version: OpenId4VCIVersion;
+    cNonce?: string;
+  },
+) {
+  if ('proof_type' in proofInput) {
+    if (opts.cNonce) {
+      throw Error(`Cnonce param is only supported when using a Proof of Posession builder`);
+      //decodeJwt(proofInput.jwt).
```
Comment (maikel-maas) [inline packages/client/lib/CredentialRequestClient.ts:45]: Remove commented-out code

File: packages/client/lib/CredentialRequestClient.ts [hunk 2]
```diff
@@ -63,11 +97,40 @@ export class CredentialRequestClient {
...
+    },
+  ): Promise<OpenIDResponse<CredentialResponse>> {
+    const transactionId = response.transaction_id;
+    const bearerToken = response.acceptance_token ?? opts?.bearerToken;
+    const deferredCredentialEndpoint = this.getDeferredCredentialEndpoint();
+    if (!deferredCredentialEndpoint) {
+      throw Error(`No deferred credential endpoint supplied.`);
+    } else if (!bearerToken) {
+      throw Error(`No bearer token present and refresh for defered endpoint not supported yet`);
+      // todo updated bearer token with new c_nonce
```
Comment (maikel-maas) [inline packages/client/lib/CredentialRequestClient.ts:123]: Is there a ticket for it?

**Review by maikel-maas — CHANGES_REQUESTED** [review]
Left small remarks

File: packages/client/lib/CredentialRequestClient.ts (hunk 2 above)
Comment (nklomp) [reply packages/client/lib/CredentialRequestClient.ts:123]: No not needed, given it throws errors for this case. So if we really hit it (not likely soon), we know what it is about to begin with

File: packages/client/lib/CredentialRequestClient.ts (hunk 1 above)
Comment (nklomp) [inline packages/client/lib/CredentialRequestClient.ts:45]:
```suggestion
```

**Pull request merged**